bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -970,4 +970,162 @@ mod tests {
         init::init();
         assert!(init::fips_mode().unwrap().is_enabled());
     }
+
+    /// Test that an application context can be set and retrieved.
+    #[test]
+    fn test_app_context_set_and_retrieve() {
+        struct TestApplicationContext {
```

Comment:
For the sake of brevity, could we use `u32` or something simple like that? If you want to use a complex type then I'd vote for something like `Vec<u32>`

Reply:
Or if not, could we just use a single declaration up one scope in the test module? Although I still think I'd like the simple types so more things would fit on one line, haha.

Reply:
That makes sense! It definitely made the tests smaller. I left the callback test as is, since it seemed worthwhile to have that test emulate what we expect users to actually use from within a callback.